account/src/main/java/com/caldevsupplychain/account/controller/AccountControllerImpl.java

```diff
@@ -0,0 +1,219 @@
+import java.util.List;
+import java.util.Optional;
+
+
+
+@Slf4j
+@RestController
+@RequestMapping("/api/account/v1")
+@AllArgsConstructor
+public class AccountControllerImpl implements AccountController {
+
+    private AccountService accountService;
+
+    private AccountWSBuilder accountWSBuilder;
+
+    /* validators */
+    private SignupValidator signupValidator;
+    private LoginValidator loginValidator;
+    private EditUserValidator editUserValidator;
+
+    private EmailService emailService;
+
+    /* exception handler for form fields and bind field errors together */
+    private FormErrorsExceptionHandler formErrorsExceptionHandler;
+
+
+    /************************************************************************************************
+     |									Account API													|
+     ************************************************************************************************/
+
+    @PostMapping("/signup")
+    public ResponseEntity<?> signup (
+            @RequestParam(required = false, defaultValue = "USER") String role,
+            @Validated @RequestBody UserWS userWS,
+            HttpServletRequest request,
+            BindingResult errors
+    )
+    {
+        // signup form validation
+        signupValidator.validate(userWS, errors);
+
+        if(errors.hasErrors()){
+            log.error("ERROR IN SIGNUP VALIDATION");
+            List<ErrorWS> errorWSList = formErrorsExceptionHandler.generateErrorWSList(errors);
+            return new ResponseEntity<>(new ApiErrorsWS(errorWSList), HttpStatus.UNPROCESSABLE_ENTITY);
+        }
+
+        // need to update after activated link, and user select specific role
+        userWS.setRole(role);
+
+        // check if user pre-exist or not
+        if (accountService.userExist(userWS.getEmailAddress())) {
+            return new ResponseEntity<>(new ApiErrorsWS(ErrorCode.ACCOUNT_EXIST.name(), "Account already registered."), HttpStatus.CONFLICT);
+        }
+
+        // convert userWS to userBean
+        UserBean userBean = new UserBean(userWS);
+
+        // success on checking, create user
+        User user = accountService.createUser(userBean);
+
+        try {
+            emailService.sendRegistrationVerificationTokenEmail(user.getEmailAddress(), user.getToken(), request);
+        } catch (MessagingException e) {
+            return new ResponseEntity<>(new ApiErrorsWS(ErrorCode.EMAIL_MESSAGING_EXCEPTION.name(), e.getMessage()), HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+        return new ResponseEntity<>(accountWSBuilder.buildUserWS(user), HttpStatus.CREATED);
+    }
+
+    // update as a form
+    @RequiresPermissions("account:update_user")
+    @PostMapping("/users/{uuid}")
+    public ResponseEntity<?> updateUser(
+            @PathVariable("uuid") String uuid,
+            @RequestParam String role,
+            @Validated @RequestBody UserWS userWS,
+            BindingResult errors
+    )
+    {
+
+        // signup form validation
```

Comment:
Need to make sure contextUser.uuid == uuid. Authenticated user can only update his own user info. contextUser should be set by shiro.